### PR TITLE
Configure the new rubocop rules

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -71,3 +68,40 @@ Style/StringLiterals:
...
+Style/HashExcept: # (new in 1.7)
+  Enabled: true
+Style/NegatedIfElseCondition: # (new in 1.2)
+  Enabled: true
+Style/NilLambda: # (new in 1.3)
+  Enabled: true
+Style/RedundantArgument: # (new in 1.4)
+  Enabled: true
+Style/SwapValues: # (new in 1.1)
+  Enabled: true
```
Comment (ndushay) [inline .rubocop.yml:107]: I find it easier to figure out what is going on when our rubocop.yml has split the rules into their uber-sections ...

**Review by ndushay — COMMENTED** [review]
one request for a change; not absolutely vital, so making this a comment.

File: .rubocop.yml (hunk 1 above)
Comment (jcoyne) [reply .rubocop.yml:107]: I'm just copying and pasting exactly what rubocop told me to paste in here.  If you like it formatted, perhaps consider a separate PR.

Comment (ndushay) [reply .rubocop.yml:107]: I understand exactly how and why you did it;  I'm suggesting that for team maintenance purposes, it's more useful if these PRs include taking another 30 seconds to put the diff groups of new rules under their headings.  I don't want to chase all over a rubocop.yml file to find the rules and our settings -- I want some sort of organization to it, especially as it gets longer and longer.

Comment (mjgiarlo) [reply .rubocop.yml:107]: This seems a reasonable request to me with more upside than downside.

**Comment by ndushay** [comment]
I'm closing this PR as it now has a conflict.   Someone will update rubocop.yml with the new rules eventually.

**Pull request closed without merging**